skiplang/prelude/src/skstore/FixedDir.sk

```diff
@@ -214,33 +214,27 @@ base class IFixedDir<T: frozen, RowRepresentation: frozen> protected {
   static fun getMetadata(
     data: readonly Vector<FixedRow<Array<File>>>,
   ): ?FixedDirMetadata {
-    if (data.isEmpty()) return None();
-    sourceDir: ?DirName = None();
-    kinds: ?Array<(Int, SQLParser.IKind, SQLParser.Type)> = None();
+    possibleResult: ?FixedDirMetadata = None();
 
     for (elt in data) {
       if (elt.value.size() != 1) return None();
       (elt.key, elt.value[0]) match {
       | (SKDB.RowKey(row1, k), row2 @ SKDB.RowValues _) ->
```

Comment:
Unrelated to this PR but this does not belong in the stdlib.

Reply:
Right. Though I don't see an easy way of both keeping this possible optimization for SKDB and moving it out of where it currently is.
Any idea @bennostein?